### DISCUSSION TITLE
Docker Image - Missing Default Ports for Embedded OpenDJ

### DISCUSSION THREAD

**Discussion by joelcharlebois** [OpenIdentityPlatform/OpenAM#827] [Q&A]
The Docker Image for OpenAM posted to Docker Hub, I am assuming by the OpenIdentityPlatform team, is missing default ports for the Embedded OpenDJ instance.

Creating a Default Configuration fails during the OpenDJ setup stage, because all of the required ports are set as minus one "-1".

The same can be seen when Creating a New Configuration, except there the user can set the ports.  

Screenshots attached.   

![Screenshot 2024-12-19 164021](https://github.com/user-attachments/assets/663b2821-bc40-45be-b5af-49295b6e9b13)
 

![Screenshot 2024-12-19 164436](https://github.com/user-attachments/assets/01a29856-b1ac-4572-9798-dd142b0ee202)

**Comment by maximthomas** [comment]
Could you provide more info, on how to reproduce the issue, such as OS and docker run command?
I've just run OpenAM Docker container with the command  below and set it up without issues.
```
docker run -h openam-01.domain.com -p 8080:8080 --name openam-01 openidentityplatform/openam
```

**Comment by maximthomas** [comment]
Please provide also your browser info and `/usr/openam/config/install.log` file from your OpenAM Docker container.

**Reply by joelcharlebois** [reply]
Hi, I got busy then was away for the holidays.  

I do not recall all of the details, but I will try again soon... 

I was using Ubuntu 24.04.1 LTS.  Browsers were Chrome and Brave.   

I believe I used a different hostname for my testing, which would get mapped to IP in the OS hosts file.  That said, the hostname used for the site shouldn't matter though right, for the default port listing to appear, to allow a user to use OpenAM for any site/domain they wish, or are there default configuration settings in the supplied Docker image only for and specific to that one example hostname? 

Thanks.

**Reply by joelcharlebois** [reply]
I tested again today creating a default configuration, the default OpenDJ ports were there this time and the default configuration completed without issue. My assumption is that I must have had a typo in the host name the first few times I tried....


I did however, notice two exceptions in the docker logs when I loaded the openam login page, looks like tomcat exceptions parsing request headers:  

![image](https://github.com/user-attachments/assets/6ae2af19-5c58-4846-a463-c9864ac15ee6)


Then at the login page or right after login, the docker logs also contained a few OpenAM security error messages, one complaining about not being able to create the debug directory, and another two that may just be info messages or otherwise complaining that the AMADMIN_MODE system property is null:

![image](https://github.com/user-attachments/assets/2fb9d624-819a-49c9-a8ab-b4377211631a)


Can all of the above be safely ignored or should something more be configured in the docker image?

Thanks.

**Comment by vharseko** [comment]
[HTTP method names must be tokens](https://stackoverflow.com/questions/38891866/when-spring-boot-starts-up-it-throws-the-method-names-must-be-tokens-exceptio)

please try [TIP: Quick OpenAM Docker Configuration From a Command Line](https://github.com/OpenIdentityPlatform/OpenAM/wiki/TIP%3A-Quick-OpenAM-Docker-Configuration-From-a-Command-Line)
